response_operations_ui/views/messages.py

```diff
@@ -115,6 +115,10 @@ def view_conversation(thread_id):
     if latest_message["unread"] and _can_mark_as_unread(latest_message):
         message_controllers.remove_unread_label(latest_message["message_id"])
 
+    if category != "SURVEY":
+        # Required to produce the correct back link for the messages
+        refined_thread[0]["survey"] = ""
```

Comment:
Realistically we should be doing this in the dict create stage, i.e line 783. To create it then change it is not something we want to do, however I see that refine is being used for other things, but I reckon it's viable in all those situations. With that being said you can leave as it is, as it is already a bit of a mess, perhaps do a nicer var rather than updating the dict.

Reply:
Good spot. I've moved this to the dictionary.